pkg/translator/loki/convert.go

```diff
@@ -26,13 +26,14 @@ import (
 )
 
 const (
-	hintAttributes = "loki.attribute.labels"
-	hintResources  = "loki.resource.labels"
+	hintAttributes     = "loki.attribute.labels"
+	hintResources      = "loki.resource.labels"
+	levelAttributeName = "level"
 )
 
 var defaultExporterLabels = model.LabelSet{"exporter": "OTLP"}
 
-func convertAttributesAndMerge(logAttrs pcommon.Map, resAttrs pcommon.Map) model.LabelSet {
+func convertAttributesAndMerge(logAttrs pcommon.Map, resAttrs pcommon.Map, severity plog.SeverityNumber) model.LabelSet {
```

Comment:
First, `convertAttributesAndMerge` now does more than its name tells us. I would favor not changing the scope of this method's functionality.
Second, we would now create a new instance of `model.LabelSet` with `convertSeverityToLabel` and then merge it with the result `model.LabelSet` of attr and res conversions, which would again result in an allocation of a new instance of `model.LabelSet`, plus 2 passes over those label sets. It seems to be a bit expensive. I was wondering if we could just modify two `log` attributes in `LogsToLoki`: 1. Set a new log `level` attribute from  `log.SeverityNumber` 2. Update the `loki.attribute.labels` with `level` hint.  If we do that, we can leave `convertAttributesAndMerge` without modifications and no additional allocations. 
Wdyt?

Reply:
Sounds good but if I do exactly as suggested, if the level hint isn't provided, then the created level attribute will stay where he is and the final entry would be `{"traceid":"01020304000000000000000000000000","attributes":{"level":8}}`, which I guess is not expected...

Reply:
Misundertood, will do that tomorrow. I guess we should préserve already existing level attribute.

Reply:
Ok I pushed something, but I find this code to contains to many specifi severity ifs....